syntax = "proto3";

package customer;

service CustomerService {
  rpc GetCustomer (GetCustomerRequest) returns (CustomerResponse);
  rpc ListCustomers (ListCustomersRequest) returns (ListCustomersResponse);
  rpc CreateCustomer (CreateCustomerRequest) returns (CustomerResponse);
  rpc DeleteCustomer (DeleteCustomerRequest) returns (DeleteCustomerResponse);
}

message Customer {
  int64 id = 1;
  string full_name = 2;
  string email = 3;
  string created_at = 4;
}

message GetCustomerRequest {
  int64 id = 1;
}

message CustomerResponse {
  Customer customer = 1;
}

message ListCustomersRequest {
  int32 limit = 1;
  int32 offset = 2;
}

message ListCustomersResponse {
  repeated Customer customers = 1;
}

message CreateCustomerRequest {
  string full_name = 1;
  string email = 2;
}

message DeleteCustomerRequest {
  int64 id = 1;
}

message DeleteCustomerResponse {
  bool success = 1;
}